dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java

```diff
@@ -154,6 +154,11 @@ private static CategoryCombo getCategoryCombo( DataSet dataset, Set<Organisation
             {
                 for ( DataElementCategory cat : cats )
                 {
+                    if ( cat.getAccess() != null && !cat.getAccess().isRead() )
```

Comment:
This FormUtils.fromDataSet is also called by DataSetController, and currently it's not checking for user authorities and also organisationUnit access. 

So if those authorities = null means that it should not check and everything should be returned..